packages/form-upload/FileRow.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'reactstrap';
+import Icon from '@availity/icon';
+import UploadProgressBar from './UploadProgressBar';
+
+const fileTypeIconMap = {
+  png: 'file-image',
+  jpg: 'file-image',
+  jpeg: 'file-image',
+  gif: 'file-image',
+  ppt: 'file-powerpoint',
+  pptx: 'file-powerpoint',
+  xls: 'file-excel',
+  xlsx: 'file-excel',
+  doc: 'file-word',
+  docx: 'file-word',
+  txt: 'doc-alt',
+  text: 'doc-alt',
+  zip: 'file-archive',
+  '7zip': 'file-archive',
+  xml: 'file-code',
+  html: 'file-code',
+  pdf: 'file-pdf',
+};
+
+const FileRow = ({ onRemove, children, file }) => {
+  const remove = () => {
+    onRemove(file.id);
+  };
+
+  const progressBar = () => <UploadProgressBar upload={file} />;
+
+  const ext = file.file.name
+    .split('.')
+    .pop()
+    .toUpperCase();
+
+  const icon = fileTypeIconMap[ext.toLowerCase()] || 'doc';
+
+  if (typeof children === 'function') {
+    return children({
+      file,
```

Comment:
Instead of having `children` as a function for the typescript definition can we add these all as named arguments to the function that returns void in the definition. This way people can intellisense and also not have errors when rendering children with a function.